test/language/statements/async-generator/yield-star-sync-next.js

```diff
@@ -0,0 +1,171 @@
+    6. Repeat
+      a. If received.[[Type]] is normal, then
+        i. Let innerResult be ? IteratorNext(iterator, received.[[Value]]).
+        ii. Let innerResult be ? Invoke(iterator, "next",
+            « received.[[Value]] »).
+        iii. If generatorKind is async, then set innerResult to
+             ? Await(innerResult).
+        ...
+        v. Let done be ? IteratorComplete(innerResult).
+        vi. If done is true, then
+           1. Return ? IteratorValue(innerResult).
+        vii. Let received be GeneratorYield(innerResult).
+      ...
+
+    GetIterator ( obj [ , hint ] )
+
+    ...
+    3. If hint is async,
+      a. Set method to ? GetMethod(obj, @@asyncIterator).
+      b. If method is undefined,
+        i. Let syncMethod be ? GetMethod(obj, @@iterator).
+        ii. Let syncIterator be ? Call(syncMethod, obj).
+        iii. Return ? CreateAsyncFromSyncIterator(syncIterator).
+    ...
+
+    %AsyncFromSyncIteratorPrototype%.next ( value )
+
+    ...
+    5. Let nextResult be IteratorNext(syncIterator, value).
+    ...
+    7. Let nextValue be IteratorValue(nextResult).
+    ...
+    9. Let nextDone be IteratorComplete(nextResult).
+    ...
+    12. Perform ! Call(valueWrapperCapability.[[Resolve]], undefined,
+        « nextValue »).
+    ...
+    14. Set onFulfilled.[[Done]] to nextDone.
+    15. Perform ! PerformPromiseThen(valueWrapperCapability.[[Promise]],
+        onFulfilled, undefined, promiseCapability).
+    ...
+
+    Async Iterator Value Unwrap Functions
+
+    1. Return ! CreateIterResultObject(value, F.[[Done]]).
+
+ flags: [async]
+---*/
+
+var log = [];
+var iter = {
+  get [Symbol.iterator]() {
+    log.push({ name: "get [Symbol.iterator]" });
+    return function(...args) {
+      log.push({
+        name: "call [Symbol.iterator]",
+        args: args
+      });
+      var nextCount = 0;
+      return {
+        get next() {
+          log.push({ name: "get next" });
+          return function(...args) {
+            log.push({
+              name: "call next",
+              args: args
+            });
+
+            nextCount++;
+            if (nextCount == 1) {
+              return {
+                get value() {
+                  log.push({ name: "get next value (1)" });
+                  return "next-value-1";
+                },
+                get done() {
+                  log.push({ name: "get next done (1)" });
+                  return false;
+                }
+              };
+            }
+
+            return {
+              get value() {
+                log.push({ name: "get next value (2)" });
+                return "next-value-2";
+              },
+              get done() {
+                log.push({ name: "get next done (2)" });
+                return true;
+              }
+            };
+          };
+        }
+      };
+    };
+  },
+  get [Symbol.asyncIterator]() {
+    log.push({ name: "get [Symbol.asyncIterator]" });
+    return null;
```

Comment:
The spec language agrees with you, this should be a TypeError trying to invoke `null`. V8 gets this wrong too, though. Maybe it would make sense to use `null` as a sentinel to use Async-from-Sync Iterator too?